File: CWE_Java/src/main/java/com/example/cwe/CWE416/NonComplaint.java
package com.example.cwe.CWE416;

//      CWE-416: Use After Free

public class NonComplaint {
    public static void main(String[] args) {

        NonComplaint obj = new NonComplaint();
        obj.getStatus();
        obj = null;
        System.gc();
        obj.getStatus();    // obj is used after it is free

    }

    void getStatus(){
    System.out.println("Method of NonComplaint Class");
    }

}

File: CWE_JavaScipt/src/CWE416/NonComplaint.js
//      CWE-416	Use After Free


function createUser() {
  let user = { name: "John", email: "john@example.com" };
  return user;
}

let user = createUser();
console.log(user.name);

user = null;
console.log(user.name); // Error: accessing property of null
